### PR TITLE
docs: add expressions to `Validate` test cases and API documentation

### LINKED ISSUE #36: Use of spdxexp as a library
👋 Thanks for the awesome implementation and `Satisfies` functionality.

Over in [syft](https://github.com/anchore/syft) we're trying to upgrade how we handle the complex license statements from SPDX.

I noticed that `parse.go` and `scan.go` has a lot of really good private functionality for reading these complex SPDX expressions. 

Would you all accept a PR that exposed a few methods of this package so we could incorporate the parsing logic into how we identify and parse license in our SBOM output?

### PR DESCRIPTION
**Pull request by spiffcs**
Closes #36 

👋  - Thanks for taking the time to look at this PR - I started this thinking [syft](https://github.com/anchore/syft) would need to leverage the `parse` function as part of this libraries public API to get `node` which would give `ReconstructLicenseString`. We're revamping some of how we do license detection/presentation and I wanted a way to be sure that if we return an expression it's valid.

I was very pleased after experimenting more that `ValidateLicenses` could be used with complex spdx expressions.

I opened issue #36 a little while ago, but think we can close it given this PR updates the docs/tests to point users to this additional use case of complex expressions as inputs to validate.

### Linting changes
When I downloaded the latest `golangci-lint` I saw a few changes that I boy scouted, but can drop that commit if you all have a different configuration for the linter (I double checked it was using the config found in the repo, but have been tripped up by this before). I noticed there was some excludes for `revive`, but don't think my changes synced with those exclusions:

Screenshot 2023-04-13 at 12 30 41 PM

Thanks for the great library!

### CODE REVIEW

File: spdxexp/scan.go [hunk 1]
```diff
@@ -269,10 +269,8 @@ func (exp *expressionStream) normalizeLicense(license string) *token {
 			return token
 		}
 	}
-	if token := deprecatedLicenseLookup(license); token != nil {
-		return token
-	}
-	return nil
+
+	return deprecatedLicenseLookup(license)
```
Comment (ajhenry) [inline spdxexp/scan.go:273]: Always love less code 😎

**Review by ajhenry — APPROVED** [review]
This look good on my end! Thanks for the contribution 🚀

File: spdxexp/satisfies_test.go [hunk 2]
```diff
@@ -20,6 +20,12 @@ func TestValidateLicenses(t *testing.T) {
 		{"Some invalid", []string{"MTI", "Apche-2.0", "GPL-2.0"}, false, []string{"MTI", "Apche-2.0"}},
 		{"GPL-2.0", []string{"GPL-2.0"}, true, []string{}},
 		{"GPL-2.0-only", []string{"GPL-2.0-only"}, true, []string{}},
+		{"SPDX Expressions are valid", []string{
+			"MIT AND APACHE-2.0",
+			"MIT AND APCHE-2.0",
+			"LGPL-2.1-only OR MIT OR BSD-3-Clause",
+			"GPL-2.0-or-later WITH Bison-exception-2.2",
+		}, false, []string{"MIT AND APCHE-2.0"}},
```
Comment (elrayle) [inline spdxexp/satisfies_test.go:28]: Excellent addition of a test for expressions.

**Review by elrayle — APPROVED** [review]
Thanks for your contribution.  It's always great to see new documentation and testing that will help the community have a better understanding of how to leverage the package.

**Pull request merged**